lib/pokerwars/game.ex

```diff
@@ -0,0 +1,74 @@
+defmodule Pokerwars.Game do
+  alias Pokerwars.{Deck}
+
+  defstruct players: [], status: :waiting_for_players, current_deck: nil, original_deck: nil
+
+  def create(deck \\ Deck.in_order) do
+    {
+      :ok,
+      %__MODULE__{ original_deck: deck }
+    }
+  end
+
+  def apply_action(game, action) do
+    phase(game.status, game, action)
+  end
+
+  defp phase(:waiting_for_players, game, action) do
+    waiting_for_players(action, game)
+  end
+
+  defp phase(:ready_to_start, game, action) do
+    ready_to_start(action, game)
+  end
+
+  defp waiting_for_players({:join, player}, game) do
+    players = game.players ++ [player]
```

Comment:
If we don't care about the order of the players we can do `players = [player | game.players]`